src/AssumeRoleProvider.ts

```diff
@@ -1,13 +1,53 @@
-import Http from 'http'
-import Https from 'https'
+import type http from 'node:http'
```

Comment:
I believe we should put the LICENSE boilerplate at the top of the file

Reply:
Do not review this pr please, review seprated amall ones

Reply:
https://github.com/minio/minio-js/pull/1119

Reply:
Oh do you want to use this PR as a base for migrating everything else into smaller chunks?

Reply:
> Oh do you want to use this PR as a base for migrating everything else into smaller chunks?

Yes, so there is no need to review this PR.

Reply:
I didn't expect this pr to me be merged.

Reply:
Alright. You should rename the PR then.